### PR TITLE
InstructionTests: update module search

### LINKED ISSUE #58: Modules in an assembly prepared by Naggum.Assembler
_(See [Naggum.Test/InstructionTests.fs](https://github.com/codingteam/naggum/pull/39/files#diff-ab898889c230e6089a5b9b875ee5191dR24).)_

Currently there are more managed modules than I expected (and I expected exactly one module). We need to check if we're doing everything right and document that behavior.

_Note:_ not sure if it a bug and/or something we need to enhance, so filed as both a bug and an enhancement.

### PR DESCRIPTION
**Pull request by ForNeVeR**
That wasn't a bug. I've found that it's a normal behavior: it really generates a new implicit "manifest module" for every dynamic assembly. That's not a problem; after we save the assembly there are no any redundant modules.

So we'll just filter the assembly modules and take the one that's named after our assembly.

Closes #58.

### CODE REVIEW

File: Naggum.Test/InstructionTests.fs [hunk 1]
```diff
@@ -12,7 +12,7 @@ let checkResult (body : string) (expectedResult : obj) =
     // TODO: FSCheck tests
     let source =
         sprintf
-        <| "(.assembly Hello
+        <| "(.assembly TestAssembly
```
Comment (rexim) [inline Naggum.Test/InstructionTests.fs]: Why do you prefer Hello over TestAssembly?

File: Naggum.Test/InstructionTests.fs [hunk 2]
```diff
@@ -21,7 +21,13 @@ let checkResult (body : string) (expectedResult : obj) =
     use stream = new MemoryStream(Encoding.UTF8.GetBytes source)
     let repr = Processor.prepare "file.ngi" stream |> Seq.exactlyOne
     let assembly = Assembler.assemble AssemblyBuilderAccess.RunAndCollect repr
-    let ``module`` = assembly.GetModules () |> Seq.last // TODO: More proper check for *our* module. Why are there 2?
+
+    // There's also a dymanic manifest module in a dynamic assembly; we need to filter that.
+    let ``module`` =
+        assembly.GetModules ()
+        |> Seq.filter (fun m -> m.ScopeName = assembly.GetName().Name)
+        |> Seq.exactlyOne
```
Comment (rexim) [inline Naggum.Test/InstructionTests.fs]: If I understand correctly filter will leave only one element. So why do you prefere exactlyOne over last? last should also work in this case, I guess.

Comment (ForNeVeR) [inline Naggum.Test/InstructionTests.fs]: `exactlyOne` will also throw an exception if it have found more than one element. It's a good situation to use it in tests so they will fall if we have something unexpected.

Comment (rexim) [inline Naggum.Test/InstructionTests.fs]: Makes sense. Thanks!

File: Naggum.Test/InstructionTests.fs (hunk 1 above)
Comment (ForNeVeR) [inline Naggum.Test/InstructionTests.fs]: Because it's more serious and enterprisey name! We're making serious shit here.

Comment (rexim) [inline Naggum.Test/InstructionTests.fs]: Ok.

**Comment by rexim** [comment]
The serious shit has been approved

**Comment by ForNeVeR** [comment]
*Seriously* approved! (seriousface)

**Comment by ForNeVeR** [comment]
smile_new_512

**very seriously staring directly into your soul**

**Comment by ForNeVeR** [comment]
And, yeah, thanks you for the review.

**Pull request merged**